lib/matplotlib/backends/backend_pdf.py

```diff
@@ -1038,9 +1039,32 @@ def _embedTeXFont(self, fontinfo):
             self._type1Descriptors[(fontinfo.fontfile, effects)] = fontdesc
         fontdict['FontDescriptor'] = fontdesc
 
+        # Use TeX Font Metrics file to get glyph widths (TeX uses its 12.20 fixed point
+        # representation and we want 1/1000 text space units)
+        tfm = fontinfo.dvifont._tfm
+        widths = [(1000 * metrics.tex_width) >> 20
+                  if (metrics := tfm.get_metrics(char)) else 0
+                  for char in range(fc, lc + 1)]
+        fontdict['Widths'] = widthsObject = self.reserveObject('glyph widths')
         self.writeObject(fontdictObject, fontdict)
+        self.writeObject(widthsObject, widths)
```

Comment:
Does this need to move after the `fontdict`? It should be reserved already.

Reply:
It doesn't matter which order we write them in, but I think the Widths object needs to be indirect and not included in the font dictionary. This does need to come after we know the font encoding, which in turn depends on the glyph set.

But I guess you mean that this would read better if we write out the widths object first? Either way works for me.

Reply:
Yes, it did used to be written first as well, but moreso that it would be along with the other width-related code.